Sources/NIOHTTPCompression/HTTPResponseDecompressor.swift

```diff
@@ -68,25 +86,27 @@ public final class NIOHTTPResponseDecompressor: ChannelDuplexHandler, RemovableC
 
             context.fireChannelRead(data)
         case .body(var part):
-            switch self.state {
-            case .compressed(_, let originalLength):
+            switch self.compression {
+            case .some(var compression):
```

Comment:
You aren't assigning the `compression` object back here, so these changes are all done to a copy which is thrown away. This is another reason to add the tests for the compression ratio policing: they'd have revealed this bug too.